src/main/java/AttendanceTracker/EnteredYear.java

```diff
@@ -1,17 +1,31 @@
 package AttendanceTracker;
 
+import java.util.InputMismatchException;
 import java.util.Scanner;
 
 public class EnteredYear {
-    static int enteredYear(){
-        Scanner scan = new Scanner(System.in);
-        System.out.print("Enter the year: ");
-        int userEnteredYear = scan.nextInt();
-        while( userEnteredYear < 2010 || userEnteredYear > 2023){
-            System.out.println("Please enter a valid year: ");
-            userEnteredYear = scan.nextInt();
+    static int enteredYear() {
+        Scanner scanner = new Scanner(System.in);
+        int userEnteredYear;
+        try{
+            do {
+                System.out.print("Enter the year: ");
+                while (!scanner.hasNextInt()) {
```

Comment:
while we can have this try catch, this will stop execution, let this code execute only if the input is an int, otherwise keep the user in the loop until what we expect is given.

Reply:
Should I keep the try catch or just get rid of it?